lib/datadog/core/telemetry/client.rb

```diff
@@ -16,12 +16,13 @@ class Client
         include Core::Utils::Forking
 
         # @param enabled [Boolean] Determines whether telemetry events should be sent to the API
-        def initialize(enabled: true)
+        # @param interval [Float] How frequently heartbeats will be reported, in seconds.
+        def initialize(heartbeat_interval:, enabled: true)
           @enabled = enabled
           @emitter = Emitter.new
           @stopped = false
           @unsupported = false
-          @worker = Telemetry::Heartbeat.new(enabled: @enabled) do
+          @worker = Telemetry::Heartbeat.new(enabled: @enabled, interval: heartbeat_interval) do
```

Comment:
Minor: I suggest propagating the `_seconds` prefix internally as well:

```suggestion
          @worker = Telemetry::Heartbeat.new(enabled: @enabled, heartbeat_interval_seconds: heartbeat_interval_seconds) do
```